### PR TITLE
[ibmswtpm2] Integrate ibmswtpm2

### PR DESCRIPTION
**Pull request by pdxjohnny**
Signed-off-by: John Andersen <john.s.andersen@intel.com>

### CODE REVIEW

File: projects/ibmswtpm2/fuzzer.cc [hunk 1]
```diff
@@ -0,0 +1,78 @@
+#include "TpmBuildSwitches.h"
```
Comment (jonathanmetzman) [inline projects/ibmswtpm2/fuzzer.cc]: Let's upstream this target at some point (as described in our docs on [ideal integrations](https://google.github.io/oss-fuzz/advanced-topics/ideal-integration/))

Comment (pdxjohnny) [reply projects/ibmswtpm2/fuzzer.cc]: Sounds good!

**Comment by jonathanmetzman** [comment]
Who are some users of ibmswtmp2?

**Comment by pdxjohnny** [comment]
This simulator shares a lot of the same codebase with the Microsoft TPM simulator. As well as the swtpm2 project which provides virtualized TPMs to QEMU guests. I'm addition to that. I can't confirm but have heard rumors that many physical TPMs are implemented by taking the simulator code and making it run on and embedded device. Which means it shares the same codebase as many of them. So I'm thinking that merging into OSS Fuzz should provide bugs which effect physical TPMs.

File: projects/ibmswtpm2/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+FROM gcr.io/oss-fuzz-base/base-builder
+MAINTAINER john.s.andersen@intel.com
+ARG SIM_DL_URL=https://astuteinternet.dl.sourceforge.net/project/ibmswtpm2/ibmtpm1332.tar.gz
```
Comment (jonathanmetzman) [inline projects/ibmswtpm2/Dockerfile]: I sometimes see sourceforge mirrors go down (but I'm not sure downloads.sourceforge.net is more reliable either).

File: projects/ibmswtpm2/no_writes.patch [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+ 
+ LNFLAGS = -ggdb 			\
+ 	-DTPM_POSIX			\
+@@ -57,7 +60,8 @@ LNFLAGS = -ggdb 			\
+ 	-lcrypto			\
+ 	-lpthread			\
+ 	-lrt				\
+-	-I.
++	-I. \
++	-DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -lFuzzingEngine
```
Comment (jonathanmetzman) [inline projects/ibmswtpm2/no_writes.patch]: Can we use `$LIB_FUZZING_ENGINE` instead?

**Comment by jonathanmetzman** [comment]
Approved. Will land once question about $LIB_FUZZING_ENGINE is answered.

**Comment by pdxjohnny** [comment]
https://travis-ci.org/google/oss-fuzz/jobs/587585424

```
Sending build context to Docker daemon  12.29kB
Step 1/10 : FROM gcr.io/oss-fuzz-base/base-builder
Get https://gcr.io/v2/oss-fuzz-base/base-builder/manifests/latest: Get https://gcr.io/v2/token?scope=repository%3Aoss-fuzz-base%2Fbase-builder%3Apull&service=gcr.io: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
docker build failed.
Running: docker build -t gcr.io/oss-fuzz/ibmswtpm2 projects/ibmswtpm2
Building project ibmswtpm2
Running command: python /home/travis/build/google/oss-fuzz/infra/helper.py build_fuzzers ibmswtpm2 --engine libfuzzer --sanitizer memory --architecture x86_64
Failed projects: ibmswtpm2
/home/travis/.travis/functions: line 553:  3652 Terminated              travis_jigger "${!}" "${timeout}" "${cmd[@]}"
```

**Comment by pdxjohnny** [comment]
Travis keeps getting network timeouts, probably needs to be restarted

**Comment by jonathanmetzman** [comment]
Taking a look

**Comment by jonathanmetzman** [comment]
> Approved. Will land once question about $LIB_FUZZING_ENGINE is answered.

Ping

**Comment by pdxjohnny** [comment]
@jonathanmetzman Sorry, yes I fixed that. Should be good to go! Thanks! :smiley:

**Comment by jonathanmetzman** [comment]
> @jonathanmetzman Sorry, yes I fixed that. Should be good to go! Thanks!

Ah my bad for not noticing.

**Comment by pdxjohnny** [comment]
No worries! Thanks!

**Pull request merged**